Marlin/src/HAL/STM32/tft/tft_spi.cpp

```diff
@@ -43,8 +43,8 @@ void TFT_SPI::init() {
   if ((spiInstance = (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_SCK_PIN),  PinMap_SPI_SCLK)) == NP) return;
   if (spiInstance != (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_MOSI_PIN), PinMap_SPI_MOSI)) return;
 
-  #if PIN_EXISTS(TFT_MISO) && TFT_MISO_PIN != TFT_MOSI_PIN
-    if (spiInstance != (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_MISO_PIN), PinMap_SPI_MISO)) return;
+  #if PIN_EXISTS(TFT_MISO)
+    if ((TFT_MISO_PIN != TFT_MOSI_PIN) && (spiInstance != (SPI_TypeDef *)pinmap_peripheral(digitalPinToPinName(TFT_MISO_PIN), PinMap_SPI_MISO))) return;
```

Comment:
Is there a special benefit to comparing these pins in code instead of in the preprocessor?

Reply:
Some SPI pins are defined as reference to analog pins instead of numbers like `#define PA7  A7`
I've run into problem on F401 development board when these definitions could not be resolved to numbers in preprocessor and caused incorrect code behavior.

Reply:
A7 is declared as `static const uint8_t` in framework-arduinoststm32\cores\arduino\pins_arduino.h:
```
#define PIN_A7       (PIN_A6 + 1)
static const uint8_t A7 = PIN_A7;
```

Reply:
> Some SPI pins are defined as reference to analog pins

Ah, that's weird since they're digital when used with SPI. Oh well!